src/app/modules/angular-slickgrid/services/__tests__/grid-odata.service.spec.ts

```diff
@@ -1539,6 +1595,62 @@ describe('GridOdataService', () => {
         expect(query).toBe(expectation);
         expect(currentFilters).toEqual(presetFilters);
       });
+
+      it('should return a query to filter a search value with a fraction of a number that is missing a leading 0', () => {
+        const expectation = `$filter=(Duration eq 0.22)`;
+        const mockColumnDuration = { id: 'duration', field: 'duration', type: FieldType.number } as Column;
+        const mockColumnFilters = {
+          duration: { columnId: 'duration', columnDef: mockColumnDuration, searchTerms: ['.22'] },
+        } as ColumnFilters;
+
+        service.init(serviceOptions, paginationOptions, gridStub);
+        service.updateFilters(mockColumnFilters, false);
+        const query = service.buildQuery();
+
+        expect(query).toBe(expectation);
+      });
+
+      it('should return a query to filter a search value with a number that contains invalid characters', () => {
```

Comment:
could you change the test description to be a bit more explicit, I think something like this should be better
`"should return a query without invalid characters to filter a search value that does contains invalid characters"`

same goes for all the other tests below as well. 
Thanks